context/app/static/js/shared-styles/dropdowns/DropdownMenuButton/DropdownMenuButton.tsx

```diff
@@ -1,21 +1,26 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import Button from '@mui/material/Button';
+import Button, { ButtonOwnProps } from '@mui/material/Button';
 
 import { useDropdownMenuStore } from 'js/shared-styles/dropdowns/DropdownMenuProvider';
 import { StyledUpIcon, StyledDownIcon } from './style';
 
-function DropdownMenuButton({ children, menuID, ...rest }) {
+interface DropdownMenuButtonProps {
+  children: React.ReactNode;
+  menuID: string;
+  variant?: ButtonOwnProps['variant'];
+}
+
+function DropdownMenuButton({ children, menuID, variant = 'outlined', ...rest }: DropdownMenuButtonProps) {
   const { menuRef, menuIsOpen, openMenu } = useDropdownMenuStore();
 
   return (
     <Button
       onClick={openMenu}
-      variant="outlined"
+      variant={variant}
       color="primary"
       aria-controls={menuIsOpen ? menuID : undefined}
       aria-haspopup="true"
-      ref={menuRef}
+      ref={menuRef as unknown as React.RefObject<HTMLButtonElement>}
```

Comment:
What prompted this?

Reply:
When I converted this component to TS, this ref produced a type error as it's a `<HTMLDivElement>` and not the `<HTMLButtonElement>` the` Button` component expects. The ref works as expected, this was just a workaround for the typing issue - if there's a better solution, I'd be happy to talk through it.

Reply:
I think it should be safe to update the type for `DropdownMenuStoreState`'s menuRef to `RefObject<HTMLElement>` to make it a bit more open/generic.

Reply:
Done - in that case, it seems that a type cast is still necessary (currently `ref={menuRef as React.RefObject<HTMLButtonElement>`}). Is this expected?

Reply:
I think you can update the type of `useDropdownMenu` to accept a generic for the ref type if needed?

Reply:
Would you mind elaborating on where this generic would be instantiated? I'm having some trouble following the logic of `useDropdownMenuStore`, and haven't been able to successfully change the ref type.